pkg/tests/validator_test.go

```diff
@@ -0,0 +1,381 @@
+package tests
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/iotaledger/hive.go/core/safemath"
+	"github.com/iotaledger/hive.go/lo"
+	"github.com/iotaledger/hive.go/log"
+	"github.com/iotaledger/hive.go/runtime/options"
+	"github.com/iotaledger/iota-core/pkg/testsuite"
+	"github.com/iotaledger/iota-core/pkg/testsuite/mock"
+	"github.com/stretchr/testify/require"
+
+	iotago "github.com/iotaledger/iota.go/v4"
+)
+
+// IOTA Mainnet Max Supply.
+const MAX_SUPPLY = iotago.BaseToken(4_600_000_000_000_000)
```

Comment:
I think it's better to use a local variable in the testsuite setup and then use the protocol params token supply.